app/models/referral.rb

```diff
@@ -0,0 +1,24 @@
+class Referral < ApplicationRecord
+  CODE_PREFIX = 'STAYHOME'.freeze
+  REWARD_VALUE = 1000
+
+  enum status: { pending: 0, rewarded: 1 }
+
+  scope :for_candidate, ->(user) { where(candidate_id: user.id) }
+  scope :for_referrer, ->(user) { where(referrer_id: user.id) }
+
+  before_create :generate_code
+
+  private
+
+  def generate_code
+    new_code = "#{CODE_PREFIX}-#{SecureRandom.hex[0, 4].upcase}"
+
+    if Referral.exists?(referrer_id: referrer_id, code: new_code)
```

Comment:
I thought an user will have a single referral code that would be shared among his peers.
It seems we are creating a new referral code for every new referral ?